File: 2-Basic-Mathematical-Knowledge-for-Algorithms/003-Sum_of_N_Integers/typescript/main.ts
import * as readline from 'readline';

const rl = readline.createInterface({
  input: process.stdin,
  output: process.stdout
});

let lineCount: number = 0;
let n: number = 0;

// 1. 入力処理
rl.on('line', (line: string) => {
  if (lineCount === 0) {
    // 1. 個数の入力処理
    n = parseInt(line.trim(), 10);
    lineCount++;
  } else {
    // 2. 配列の入力処理
    const a: number[] = line.trim().split(' ').map(s => parseInt(s, 10));

    // 3. 合計計算処理
    const sum: number = a.reduce((acc: number, num: number) => acc + num, 0);

    // 4. 出力処理
    console.log(sum);

    rl.close();
  }
});

File: 2-Basic-Mathematical-Knowledge-for-Algorithms/003-Sum_of_N_Integers/rust/src/main.rs
use std::io;

fn main() {
    // 1. 個数の入力処理
    let mut input = String::new();
    io::stdin().read_line(&mut input).expect("Failed to read line");
    let n: usize = input.trim().parse().expect("Please enter a number");

    // 2. 配列の入力処理
    input.clear();
    io::stdin().read_line(&mut input).expect("Failed to read line");
    let a: Vec<i32> = input
        .trim()
        .split_whitespace()
        .map(|s| s.parse().expect("Please enter a number"))
        .collect();

    // 3. 合計計算処理
    let sum: i32 = a.iter().sum();

    // 4. 出力処理
    println!("{}", sum);
}